src/application/common/command.ts

```diff
@@ -15,9 +15,9 @@ export interface Command {
     iconClass?: string;
 }
 export interface CommandHandler {
-    execute(): any;
-    isEnabled(): boolean;
-    isVisible?(): boolean;
+    execute(...args: any[]): any;
+    isEnabled(...args: any[]): boolean;
```

Comment:
agreed with optional, but with overloads, an using side looks better in simple cases, e.g.
```
commands.registerCommand(APPLY_WORKSPACE_EDIT, (changes: WorkspaceEdit) =>
    !!this.workspace.applyEdit && this.workspace.applyEdit(changes)
)
```
against
```
commands.registerCommand({ id: APPLY_WORKSPACE_EDIT }, {
    execute: (changes: WorkspaceEdit) =>
        !!this.workspace.applyEdit && this.workspace.applyEdit(changes)
})
```